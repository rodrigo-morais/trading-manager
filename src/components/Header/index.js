import React from 'react'
import styled from 'styled-components'

import DropDown from '../shared/DropDown'


const Container = styled.div`
  display: inline-grid;
  grid-template-columns: 26.66% 26.66% 26.66%;
  justify-content: space-evenly;
  min-width: 100%;
`

const DropDownLeft = styled(DropDown)`
  grid-column-start: 1;
  grid-column-end: 1;
  justify-self: start;
`
const DropDownMiddle = styled(DropDown)`
  grid-column-start: 3;
  grid-column-end: 3;
  justify-self: center;
`

const DropDownRight = styled(DropDown)`
  grid-column-start: 6;
  grid-column-end: 6;
  justify-self: end;
`

const Header = () => (
  <Container>
    <DropDownLeft label="Ativo" options={['WINM20', 'WDON20']} />
    <DropDownMiddle label="Tipo" options={['Pregão', 'Abertura']} />
    <DropDownRight label="Estratégia" options={['Topo / Fundo', 'Fecha Fora / Fecha Dentro', '3x1', 'Fabolous 4']} />
  </Container>
)

export default Header
