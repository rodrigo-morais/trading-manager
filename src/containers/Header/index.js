import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import DropDown from '../../components/shared/DropDown'


const Container = styled.div`
  display: inline-grid;
  grid-template-columns: 32% 32% 32%;
  justify-content: space-evenly;
  min-width: 100%;
`

const Header = ({ trades, className, selectStock, selectType, selectStrategy, disabled }) => (
  <Container className={className}>
    <DropDown label="Ativo" options={['WINM20', 'WDON20']} value={trades.stock} onChange={selectStock} disabled={disabled} />
    <DropDown label="Tipo" options={['Pregão', 'Abertura']} value={trades.type} onChange={selectType} disabled={disabled} />
    <DropDown label="Estratégia" options={['Topo / Fundo', 'Fecha Fora / Fecha Dentro', '3x1', 'Fabolous 4']} value={trades.strategy} onChange={selectStrategy} disabled={disabled} />
  </Container>
)

Header.defaultProps = {
  className: '',
  disabled: true,
}

Header.propTypes = {
  trades: PropTypes.object.isRequired,
  className: PropTypes.string,
  selectStock: PropTypes.func.isRequired,
  selectType: PropTypes.func.isRequired,
  selectStrategy: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
}

export default Header
