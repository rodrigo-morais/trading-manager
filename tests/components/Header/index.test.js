import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Header from '../../../src/components/Header'

describe('Header', () => {
  describe('Rendering', () => {
    const wrapper = (
      <Header />
    )

    test('renders according to design', () => {
      const component = renderer.create(wrapper)
      const tree = component.toJSON()

      expect(tree).toMatchSnapshot()
    })

    test('first label as "Ativo"', () => {
      const component = mount(wrapper)
      expect(component.find('label').at(0).text()).toEqual('Ativo:')
    })

    test('second label as "Tipo"', () => {
      const component = mount(wrapper)
      expect(component.find('label').at(1).text()).toEqual('Tipo:')
    })

    test('second label as "Estratégia"', () => {
      const component = mount(wrapper)
      expect(component.find('label').at(2).text()).toEqual('Estratégia:')
    })
  })
})
