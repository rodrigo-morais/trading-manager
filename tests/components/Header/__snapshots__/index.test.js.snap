// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Rendering renders according to design 1`] = `
<div
  className="sc-bxivhb gZTJwe"
>
  <div
    className="sc-bdVaJa dzArWz"
  >
    <label
      className="sc-bwzfXH jnEWcg"
      htmlFor="Ativo"
    >
      Ativo:
    </label>
    <select
      className="sc-htpNat dlmdsH"
      disabled={false}
      id="Ativo"
      name="Ativo"
    >
      <option
        value="WINM20"
      >
        WINM20
      </option>
      <option
        value="WDON20"
      >
        WDON20
      </option>
    </select>
  </div>
  <div
    className="sc-bdVaJa dzArWz"
  >
    <label
      className="sc-bwzfXH jnEWcg"
      htmlFor="Tipo"
    >
      Tipo:
    </label>
    <select
      className="sc-htpNat dlmdsH"
      disabled={false}
      id="Tipo"
      name="Tipo"
    >
      <option
        value="Pregão"
      >
        Pregão
      </option>
      <option
        value="Abertura"
      >
        Abertura
      </option>
    </select>
  </div>
  <div
    className="sc-bdVaJa dzArWz"
  >
    <label
      className="sc-bwzfXH jnEWcg"
      htmlFor="Estratégia"
    >
      Estratégia:
    </label>
    <select
      className="sc-htpNat dlmdsH"
      disabled={false}
      id="Estratégia"
      name="Estratégia"
    >
      <option
        value="Topo / Fundo"
      >
        Topo / Fundo
      </option>
      <option
        value="Fecha Fora / Fecha Dentro"
      >
        Fecha Fora / Fecha Dentro
      </option>
      <option
        value="3x1"
      >
        3x1
      </option>
      <option
        value="Fabolous 4"
      >
        Fabolous 4
      </option>
    </select>
  </div>
</div>
`;
